dot/rpc/modules/state_integration_test.go

```diff
@@ -316,50 +317,73 @@ func TestStateModule_QueryStorage(t *testing.T) {
 		require.Error(t, err, "the start block hash cannot be an empty value")
 	})
 
-	t.Run("When coreAPI QueryStorage returns error", func(t *testing.T) {
-		coreapimock := new(mocks.CoreAPI)
-		coreapimock.On("QueryStorage", mock.AnythingOfType("common.Hash"), mock.AnythingOfType("common.Hash")).
-			Return(nil, errors.New("problem while querying"))
-
+	t.Run("When blockAPI returns error", func(t *testing.T) {
+		mockError := errors.New("mock test error")
+		ctrl := gomock.NewController(t)
+		mockBlockAPI := NewMockBlockAPI(ctrl)
+		mockBlockAPI.EXPECT().GetBlockByHash(common.Hash{1, 2}).Return(nil, mockError)
 		module := new(StateModule)
-		module.coreAPI = coreapimock
+		module.blockAPI = mockBlockAPI
 
 		req := new(StateStorageQueryRangeRequest)
 		req.StartBlock = common.NewHash([]byte{1, 2})
 
 		var res []StorageChangeSetResponse
 		err := module.QueryStorage(nil, req, &res)
-		require.Error(t, err)
-		coreapimock.AssertCalled(t, "QueryStorage", mock.AnythingOfType("common.Hash"), mock.AnythingOfType("common.Hash"))
+		assert.ErrorIs(t, err, mockError)
 	})
 
 	t.Run("When QueryStorage returns data", func(t *testing.T) {
-		blockhash := common.NewHash([]byte{123})
-
-		changes := map[common.Hash]core.QueryKeyValueChanges{
-			blockhash: core.QueryKeyValueChanges(map[string]string{
-				"0x80": "value",
-				"0x90": "another value",
-			}),
+		k1 := "0x90"
+		v1 := common.BytesToHex([]byte(`value`))
+		k2 := "0x80"
+		v2 := common.BytesToHex([]byte(`another value`))
+		expectedChanges := [][2]*string{{&k1, &v1}, {&k2, &v2}}
```

Comment:
nit with two test helper functions (you can have them in this file or in a `helpers_test.go` file):

```go
func stringToHex(s string) (hex string) {
	return common.BytesToHex([]byte(s))
}

func makeChange(keyHex, valueHex string) [2]*string {
	return [2]*string{&keyHex, &valueHex}
}
```

You could make this a bit nicer with:

```suggestion
		expectedChanges := [][2]*string{
			makeChange("0x90", stringToHex("value")),
			makeChange("0x80", stringToHex("another value")),
		}
```